desktop/apps/editorial_features/components/gucci/components/section/section.jsx

```diff
@@ -7,20 +7,31 @@ import { SectionVideo } from './section_video.jsx'
 import { SectionText } from './section_text.jsx'
 
 export const Section = (props) => {
-  const { section, onScrollOver } = props
+  const {
+    section,
+    onEnterSection,
+    onLeaveSection,
+    index
+  } = props
 
   return (
     <SectionContainer>
       <SectionVideo section={section} />
-      <Waypoint onEnter={onScrollOver} />
       <SectionText section={section} />
+      <Waypoint
+        onEnter={(waypointData => onEnterSection(index, waypointData))}
+        onLeave={(waypointData => onLeaveSection(index, waypointData))}
+        bottomOffset='50px'
+      />
```

Comment:
We do a similar thing in standard article InfiniteScroll -- this ensures that we change `activeSection` on the borders of each section and not in the middle between `SectionVideo` and `SectionText`.

Also by tracking both `onEnter` and `onLeave` (and determining the direction with Waypoint's `previousPosition` and `currentPosition`), we can change states at the top and bottom of the viewport instead of just at the bottom.